app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_instances = [Person(person["name"], person["age"]) for person in people]
+
+    for person in people:
+        person_instance = Person.people[person["name"]]
+        if "wife" in person and person["wife"] is not None:
+            person_instance.wife = Person.people[person["wife"]]
+        if "husband" in person and person["husband"] is not None:
+            person_instance.husband = Person.people[person["husband"]]
```

Comment:
The code assumes that the 'wife' and 'husband' keys are always present in the input dictionary. If these keys are missing, it could lead to a KeyError. Consider using the `get` method to safely access these keys, e.g., `person.get('wife')`.